src/react/reconcilation.js

```diff
@@ -160,29 +207,22 @@ export class Reconciler {
         if (error.name === "Invariant Violation") {
           throw error;
         }
+
+        let message;
         if (error instanceof ExpectedBailOut) {
-          this.logger.logWarning(
-            componentType,
-            `__optimizeReactComponentTree() React component tree failed due expected bail-out - ${error.message}`
-          );
-          evaluatedRootNode.message = error.message;
-          evaluatedRootNode.status = "BAIL-OUT";
+          message = `Failed to optimize React component tree for "${
+            evaluatedRootNode.name
+          }" due to an expected bail-out: ${error.message}`;
         } else if (error instanceof AbruptCompletion) {
-          this.logger.logWarning(
-            componentType,
-            `__optimizeReactComponentTree() React component tree failed due runtime runtime exception thrown`
-          );
-          evaluatedRootNode.status = "ABRUPT_COMPLETION";
+          message = `Failed to optimize React component tree for "${
+            evaluatedRootNode.name
+          }" due to an exception thrown during render phase`;
         } else {
```

Comment:
Seems like when we have a Prepack bug (not even an invariant, but a JS error like accessing a property on undefined), we get here. This doesn't seem great because we lose the original call stack. For example I got a message like this:

```
Error: Failed to optimize React component tree for "FeedStoryUFISummary" due to evaluation error: Cannot read property '$Get' of undefined
    at renderComponentTree (/Users/gaearon/p/prepack/src/react/reconcilation.js:225:15)
    at Realm.evaluateForEffects (/Users/gaearon/p/prepack/src/realm.js:761:15)
```

Could we preserve the stack, or better, only "wrap" errors that are known to be intentional (FatalError?), and aren't just bugs in Prepack?

Reply:
Good point, I'll do that for sure.